### ISSUE TITLE
Enhancement: prebuilt binaries for linuxmusl-arm64

### ISSUE THREAD

**Issue by lovell** [lovell/farmhash#48]
Future possible enhancement to provide prebuilt binaries for Linux (glibc and musl) running on ARM64.

CircleCI can be used for this purpose e.g. https://github.com/lovell/sharp/blob/main/.circleci/config.yml

<hr>

I don't have any immediate plans to work on this particular task so if anyone requires this urgently then I am available for paid consulting. Anyone wishing to support open source software development by purchasing 1 day of my time to work on this task is very welcome to do so via https://buy.stripe.com/6oE00r1Qe7sgeli6oR - thanks in advance.

**Comment by wodCZ** [comment]
Please do 🙏 

Got `farmhash@3.3.1` as a new dependency of `firebase-admin@12.1.0`, and my build on alpine docker on Apple Silicon (M2) fails because of missing python.

Getting this prebuild-install warning:
```
No prebuilt binaries found (target=3 runtime=napi arch=arm64 libc=musl platform=linux)
```

**Comment by busybox11** [comment]
You're getting this because you're using Docker. Native Apple Silicon prebuilt binary exists. You can either bundle build tools in your image for Linux or straight up use MacOS instead.

**Comment by wodCZ** [comment]
@busybox11 Thanks for the response 🙏 

I use Docker in production (built & running on linux amd64, working just fine there 👍). I just can't build the image locally to verify it works, which is not a blocker.

I suppose I could also install python and other dependencies in the image as a temporary workaround, but again, it builds fine on target environment, so that would be a waste. 
I personally can just wait it out until someone knowledgeable and kind adds the prebuilt binaries. I guess I just wanted to let the OP know that the library is now used in `firebase-admin` (1.5M downloads/wk) and eventually more users with setup similar to mine will show up in this issue.

Use Docker they said. No more "works on my machine" they said. 🤷‍♂️

**Comment by busybox11** [comment]
Yes, this is what i meant! I also had to add C++ / Python build tools in my Docker image to build my Node project. But now it works flawlessly on my ARM CI server. And I also have a cache of the built deps so that I don't have to rebuild every time, I only lose like 2-3 seconds per build compared to before. It's not a big deal, although a little annoying indeed.

To be fair, it's not Docker's fault! It works as expected in this case, it doesn't work on Linux ARM, it won't work on any other Linux ARM out of the box either haha. At least it makes it replicable in a barebones environment like an Alpine image.

Now that I think of it you could run the Docker build by specifying the `amd64` architecture, which would probably 1:1 replicate your production environment. Just use the `--platform linux/amd64` CLI flag when building. The error should disappear since it would use Rosetta and the prebuilt binary for x86 which already exists. A step further in making sure your Docker image works as expected on x86 as well!

**Comment by lovell** [comment]
> I guess I just wanted to let the OP know that the library is now used in `firebase-admin` (1.5M downloads/wk) and eventually more users with setup similar to mine will show up in this issue.

Yes, this was unexpected, and I created this issue in response to learning of this new dependee. I don't have any immediate plans to work on this particular task so if anyone requires this urgently then I am available for paid consulting.

https://github.com/lovell/farmhash/blob/ad36182c73f4d40a67166edf0b354733b09a9bfe/package.json#L4

**Comment by adarshmadrecha** [comment]
I also landed here because my `npm i` command failed on `ARM` based `Ubuntu 22.04` on aws. after upgrading from firebase-admin 12.0.0 to 12.1.0
Please add pre built binaries.

**Comment by lhermann** [comment]
In case others are looking for the same issue: In my case the `node:18-alpine` image produced the error while the `node:18` image worked.

**Comment by lovell** [comment]
v5.0.0 is now available with additional support for linuxmusl-arm64.

This major version increment contains a complete overhaul of installation. The [farmhash](https://www.npmjs.com/package/farmhash) package is now auto-published to npm with provenance and contains everything required to run on multiple platforms, without any install-time scripts.

As a nice bonus the installation size is ~30% smaller than the previous 4.x.x versions.

https://packagephobia.com/result?p=farmhash
